From d7068c6878b52c382b40e543e1b7be5afae45c96 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Wed, 16 Feb 2022 17:01:54 +0000
Subject: GCC: DecodeTimestamp::InMilliseconds cannot be constexpr.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix bug error in GCC because of using a non-constexpr inside a
constexpr. As making TimeDelta::InMilliseconds makes the binary
significantly bigger, this just removes constexpr for
DecodeTimestamp::InMilliseconds.

../../media/base/stream_parser_buffer.h:96:71: error: call to non-‘constexpr’ function ‘int64_t base::TimeDelta::InMilliseconds() const’
   96 |   constexpr int64_t InMilliseconds() const { return ts_.InMilliseconds(); }
      |                                                     ~~~~~~~~~~~~~~~~~~^~

Bug: 819294
Change-Id: I6d0fb6de7d42c857863827a568059f860505944f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3453082
Reviewed-by: Ted (Chromium) Meyer <tmathmeyer@chromium.org>
Commit-Queue: José Dapena Paz <jdapena@igalia.com>
Cr-Commit-Position: refs/heads/main@{#971898}
---
 media/base/stream_parser_buffer.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media/base/stream_parser_buffer.h b/media/base/stream_parser_buffer.h
index b37eec6470df2..9e4462905ff1d 100644
--- a/media/base/stream_parser_buffer.h
+++ b/media/base/stream_parser_buffer.h
@@ -93,7 +93,7 @@ class DecodeTimestamp {
   }
 
   constexpr double InSecondsF() const { return ts_.InSecondsF(); }
-  constexpr int64_t InMilliseconds() const { return ts_.InMilliseconds(); }
+  int64_t InMilliseconds() const { return ts_.InMilliseconds(); }
   constexpr int64_t InMicroseconds() const { return ts_.InMicroseconds(); }
 
   constexpr bool is_inf() const { return ts_.is_inf(); }
-- 
2.32.0

